tests/filename_helper_ut/filename_helper_ut.c

```diff
@@ -0,0 +1,310 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+
+#include <stdlib.h>
+#include <stdint.h>
+#include <stdbool.h>
+#include <stdarg.h>
+
+
+#include "macro_utils/macro_utils.h"
+
+#include "testrunnerswitcher.h"
+#include "umock_c/umock_c.h"
+#include "umock_c/umocktypes_charptr.h"
+#include "umock_c/umock_c_negative_tests.h"
+
+#include "c_pal/interlocked.h" /*included for mocking reasons - it will prohibit creation of mocks belonging to interlocked.h - at the moment verified through int tests - this is porting legacy code, temporary solution*/
+
+#define ENABLE_MOCKS
+#include "c_pal/gballoc_hl.h"
+#include "c_pal/gballoc_hl_redirect.h"
+#include "c_pal/string_utils.h"
+#undef ENABLE_MOCKS
+
+#include "real_gballoc_hl.h"
+#include "real_string_utils.h"
+
+#include "c_util/filename_helper.h"
+
+/*following function cannot be mocked because of variable number of arguments:( so it is copy&pasted here*/
+char* sprintf_char_function(const char* format, ...)
+{
+    char* result;
+    va_list va;
+    va_start(va, format);
+    result = vsprintf_char(format, va);
+    va_end(va);
+    return result;
+}
+
+static unsigned char va_list_to_string[] = "some va_list stringification";
+static char* umockvalue_stringify_va_list(const void* value)
+{
+    (void)value;
+    char* result = malloc(sizeof(va_list_to_string));
+    ASSERT_IS_NOT_NULL(result);
+    memcpy(result, va_list_to_string, sizeof(va_list_to_string));
+    return result;
+}
+
+static int umockvalue_are_equal_va_list(const void* left, const void* right)
+{
+    (void)memcmp(left, right, sizeof(va_list));
+    return 0;
+}
+
+static int umockvalue_copy_va_list(void* destination, const void* source)
+{
+    (void)memcpy(destination, source, sizeof(va_list));
+    return 0;
+}
+
+static void umockvalue_free_va_list(void* value)
+{
+    (void)value;
+}

```

Comment:
This is copy/paste from `flags_to_string_ut.c`.  This should maybe be a separate module.